lib/query/lib/createGraph.js

```diff
@@ -82,6 +84,23 @@ export function createNodes(root) {
     }
 }
 
+function sortFieldNames(body) {
+    const bodyCompareFunction = function (key, value) {
+        // push fields first
```

Comment:
Not sure if sorting is the right idea here. It may fail for nested links. I think the solution is to not mark the field as scheduled for deletion, or use it above the link.

Reply:
Yeah, with that sorting function we are doing kind of a guesswork. Why not make this process 2-pass? First, we determine what are props, fields, reducers and links. Second, we add them. Props and fields first, then links and reducers.

Reply:
Thank you for your time reviewing
Since the fields are not yet added in `_shouldCleanStorage`, the `hasField` method cannot see that the field is supposed to be there.
We could update[ this line ](https://github.com/cult-of-coders/grapher/blob/842dbb1ed313ecd1abdd7fc79378be673445df1b/lib/query/nodes/collectionNode.js#L248) and change it with 

```javascript
result = !this.hasField(node.linkStorageField, true) && !(node.linkStorageField in node.parent.body)
```

Would that be a better solution @theodorDiaconu ?